transaction/transactor.go

```diff
@@ -197,3 +210,13 @@ func genTID() (tid string) {
 	}
 	return
 }
+
+func GetDeviceId(r *http.Request) string {
+	vars := mux.Vars(r)
+	id, ok := vars["deviceid"]
+	if !ok {
+		//TODO: edit log statement
+		fmt.Println("id is missing in parameters")
```

Comment:
If the id isn't present I think we should add a deviceid of `mac:000000000000` which is both a valid address, but also a reserved value that should never be used.  This will allow us to track this happened easier than nothing being present.

Reply:
I'm not sure how difficult it would be to make this configurable (ideal).